### ISSUE TITLE
[Bug]: v5.22 Fail to sync with WebDAV method (5.22版本WebDAV方法同步失败)

### ISSUE THREAD

**Issue by huizhiw** [remotely-save/remotely-save#830]
### What happened?

I use the WebDAV to sync my notes cross windows/IPad. After I update the plugin to v5.22 from 5.20, the plugin fail to sync, but I check the connection between plugin and server (I use Synology), the connection is working. 

I downgrade the version to 5.20 now and it works well again. I guess there is a bug? or some more settings need to be changed in the plugin?

Appreciate your work! This is the best sync plugin in Obsidian at present.

---
我用WebDAV同步笔记的，使用的是群晖的webdav。
5.20版本的插件没问题，5.22以后同步失败，但是检查连接可以连上群晖。可能需要增加额外设置还是bug?

目前降级到5.20在使用，没有任何问题。

十分感谢作者的付出，这是目前Obsidian上webdav同步最好的插件。

### What OS are you using?

Windows

### What remote cloud services are you using? (Please choose the specified one if it's in the list)

webdav, webdav (ownCloud)

### Version of the plugin

5.22

### Version of Obsidian

1.6.7

### Using password or not

- [X] Yes.

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by fyears** [comment]
could you provide the screenshot of the errors?

**Comment by fyears** [comment]
have you set any deny list or allow list?

**Comment by Lanjelin** [comment]
Running 0.4.19 on one client, 0.5.23 on another.  
Accessing the same dav server on both, the 0.4.19 works as it's always done, while 0.5.23 throws 403 errors.  

The dav-server is configured as follows: 
 - https://dav.example.org/ should thow 403
 - https://dav.example.org/username would allow user to authenticate

The vault itself is then stored on https://dav.example.org/username/vault-name

Changing Base Directory to `username` on 0.5.23, and it will authenticate, but this is not desired as there are other files stored on the path.

**Comment by fyears** [comment]
> Changing Base Directory to username on 0.5.23, and it will authenticate, but this is not desired as there are other files stored on the path.

though i want to debug it, could you provide a detailed reproducible way to setup the server? a docker file is preferred.

**Comment by Lanjelin** [comment]
I updated the version of remotely-save along with moving my setup between servers, and it turned out nginx was missing write-permission to the dav directories.  

I'll include the gist of how I'm running it though, but I guess there's been added a check in a later version that remotely-save can actually write to its given path.  

#### docker-compose.yaml
```yaml
services: 
  nginx:
    container_name: nginx
    image: lscr.io/linuxserver/nginx
    environment:
      - UMASK=002
      - PUID=99
      - PGID=100
    volumes:
      - ./nginx/config/nginx:/config
      - ./nginx/dav:/dav
    networks:
      - web

  cloudflared:
    container_name: cloudflared
    image: cloudflare/cloudflared:latest 
    command: tunnel --no-autoupdate run
    environment:
      - TUNNEL_TOKEN=[redacted]
    restart: unless-stopped
    networks:
      - web

networks:
  web:
    driver: bridge
```

#### config/nginx/nginx/site-confs/dav.conf
```yaml
# Declare the safe cross-origin hosts
map $http_origin $cors_origin_header {
  hostnames;
  default "https://example.org";
  "https://kw.example.org" "$http_origin";
  "https://kw.example.com" "$http_origin";
}
# Declare CORS exposed response headers
map $host $std_response_headers {
  default "Content-Type, Content-Range, Content-Language, Date, Content-Length, Content-Encoding";
}
map $host $cache_control_response_headers  {
  default "Etag, Last-Modified";
}
map $host $dav_response_headers {
  default "Dav";
}
map $host $cors_expose_headers {
  default "${dav_response_headers}, ${std_response_headers}, ${cache_control_response_headers}";
}
# Declare CORS allowed request headers
map $host $std_request_headers {
  default "Authorization, Origin, X-Requested-With, Range, Accept-Encoding, Content-Length, Content-Type";
}
map $host $dav_request_headers {
  default "If-Match, If-None-Match, If-Modified-Since, Depth";
}
map $host $keepass_allow_headers {
  # For KeeWeb
  default "DNT, X-CustomHeader, Keep-Alive, User-Agent, Cache-Control, X-Accept-Charset, X-Accept, Accept, Destination, Overwrite";
}
map $host $cors_allow_headers {
  default "${dav_request_headers}, ${std_request_headers}, ${keepass_allow_headers}";
}
# Detect a preflight request
map $http_access_control_request_headers $preflight_h {
  default "true";
  "" "false";
}
map $http_access_control_request_method $preflight_m {
  default "true";
  "" "false";
}
map $request_method $preflight {
  default "false";
  "OPTIONS" "${preflight_h}${preflight_m}true";
}

server {
  server_name dav.example.org;
  listen 443 ssl;
  http2 on;
  include /config/nginx/ssl.conf;
  root /dav; # Mounted to ./nginx/dav 
  # autoindex on;

  client_max_body_size 250M;
  charset UTF-8;
 
  # add_header Access-Control-Allow-Origin *;
  add_header "Access-Control-Allow-Origin" "$cors_origin_header" always;
  add_header "Access-Control-Allow-Credentials" "true" always;
  add_header "Access-Control-Expose-Headers" "$cors_expose_headers" always;
  location / {
    # add_header Access-Control-Allow-Origin *;
    # auth_basic               "Restricted";
    # auth_basic_user_file     /config/nginx/auth/las_e.htpasswd;
    # dav_methods              PUT DELETE MKCOL COPY MOVE;
    # dav_ext_methods          PROPFIND OPTIONS;
    # dav_access               user:rw group:rw;
    
    # create_full_put_path     on;
    # autoindex on;
  
    if ($preflight = "truetruetrue"){
      add_header "Access-Control-Allow-Origin" "$cors_origin_header";
      add_header "Access-Control-Allow-Headers" "$cors_allow_headers";
      add_header "Access-Control-Expose-Headers" "ETag"; # For KeeWeb
      add_header "Access-Control-Allow-Methods" "PROPFIND, COPY, MOVE, MKCOL, CONNECT, DELETE, DONE, GET, HEAD, OPTIONS, PATCH, POST, PUT, LOCK, UNLOCK";
      add_header "Access-Control-Max-Age" 1728000;
      add_header "Content-Type" "text/plain charset=UTF-8";
      add_header "Content-Length" 0;
      return 204;
    }
  }

  location /username {
    # add_header Access-Control-Allow-Origin *;
   
    if ($preflight = "truetruetrue"){
      add_header "Access-Control-Allow-Origin" "$cors_origin_header";
      add_header "Access-Control-Allow-Headers" "$cors_allow_headers";
      add_header "Access-Control-Expose-Headers" "ETag"; # For KeeWeb
      add_header "Access-Control-Allow-Methods" "PROPFIND, COPY, MOVE, MKCOL, CONNECT, DELETE, DONE, GET, HEAD, OPTIONS, PATCH, POST, PUT, LOCK, UNLOCK";
      add_header "Access-Control-Max-Age" 1728000;
      add_header "Content-Type" "text/plain charset=UTF-8";
      add_header "Content-Length" 0;
      return 204;
    }

    auth_basic               "Restricted";
    auth_basic_user_file     /config/nginx/auth/username.htpasswd;
    dav_methods              PUT DELETE MKCOL COPY MOVE;
    dav_ext_methods          PROPFIND OPTIONS;
    dav_access               user:rw group:rw;
     
    create_full_put_path     on;
    autoindex on;

    # error_log                /dev/stdout;
  }
}

```

**Comment by huizhiw** [comment]
> could you provide the screenshot of the errors?

I found that I had the same problem as the issue 828 and it has been fixed in 5.23 version. Now I update the version to 5.24 and everything works well!

Thanks again for your work! Appreciate it!

**Comment by fyears** [comment]
ok, so the issues are solved

moreover, checking connections try to create and upload files, so if the server blocks the creation the process will fail
